### ISSUE TITLE
Thank you for your attention. Welcome any questions here!

### ISSUE THREAD

**Issue by JianGuanTHU** [JianGuanTHU/StoryEndGen#1]

**Comment by BinbinBian** [comment]
Hi JianGuan, I am interested in your work. Could you share me your data pre-processing code?  Especially the script how do you process the conceptnet to build your training data . Thanks for sharing!

**Comment by JianGuanTHU** [comment]
@BinbinBian I'preparing the code. I will send the code to you as soon as possible.

**Comment by qywu** [comment]
Hi JianGuan, I am interested in your work, but can't reproduce the results you have. I am using the default hyper-parameters. Do you use another set of hyper-parameters during the training?

The attached is the output during the training. I used nltk to calculate BLEU-1. The resulting BLEU-1 is around 0.21 for the model which has val_set perplexity 9.51.

[train.txt](https://github.com/JianGuanTHU/StoryEndGen/files/2967429/train.txt)

**Comment by TobiasLee** [comment]
Hi JianGuan, thanks for open-sourcing your code. 
You mentioned in README:
> Note that there is something wrong with bleu-2 in the result reported in the paper. 

 Could you please provide more details, like what's the problem?

**Comment by JianGuanTHU** [comment]
@qywu I guess it maybe result from the issue of dataset splits.  ROCStories corpus contains 98,162 short stories as training data, and an additional 1,817 stories for validation and test, respectively. Originally, we had intended to regard the titles as part of input, and since only the training set of the ROCStories corpus contains titles, we ignore the additional data. But as you can see in the paper, we just input the first four sentences into our model and for convenience we still use only the training data. However, the additional ~3700 stories can actually be employed for testing, so they are used in the shared code. Anyway, in the paper we extract the first 90,000 stories for training and the rest (8,162 stories) for testing. I will clarify this point in Readme as soon as possible. And if convenient you can send the generation of the result so that I can help analyse the issue. Thank you!

**Comment by JianGuanTHU** [comment]
@TobiasLee I mean that we  have mistaken the bleu score for modified n-gram precision score in the official publication. But we have update it in arxiv preprint version. Thank you!

**Comment by qywu** [comment]
@JianGuanTHU 

Thank you for your response, but there is still confusion.

Are you using a different dataset (different split) which is not shared in your code? For the additional 3700 stories, are you referring to Cloze?

By the way, I have listed the results I have for the dataset provided in your code and also for Cloze:

```
Dataset     | Size  | BLEU-1 | BLEU-2
data/test.* | 4081 | 0.2063   | 0.035
data/val.*  | 4081 | 0.2077   | 0.0347
Cloze        | 3742 | 0.1975   | 0.0315
```


I guess the problem might be the calculation of BLEU-1, BLEU-2.
I am using nltk.sentence_bleu(xxx, weights=(1, 0, 0 ,0)) for BLEU-1, and
nltk.sentence_bleu(xxx, weights=(0.5, 0.5, 0 ,0)) for BLEU-2, 
could it be that you used something different?

*I have attached outputs for test and validation.
[output_validation.txt](https://github.com/JianGuanTHU/StoryEndGen/files/2998202/output_validation.txt)
[output_test.txt](https://github.com/JianGuanTHU/StoryEndGen/files/2998203/output_test.txt)

**Comment by qywu** [comment]
@JianGuanTHU @TobiasLee  Could it be possible when calculating the BLEU score, you used n-gram Precision instead? I have used nltk's modified precision, and finally get similar results for BLEU-1, BLEU-2.

nltk.translate.bleu_score.modified_precision

Could you share the code of the metric you used for calculating BLEU-1, BLEU-2?

**Comment by TobiasLee** [comment]
> @TobiasLee I mean that we have mistaken the bleu score for modified n-gram precision score in the official publication. But we have update it in arxiv preprint version. Thank you!

@JianGuanTHU Thanks for the clarification. So the current results shown in the README.md and the [Arxiv](https://arxiv.org/pdf/1808.10113.pdf) version are correct, am I right?

**Comment by JianGuanTHU** [comment]
@TobiasLee Yes.

**Comment by vinayshekharcmu** [comment]
Hi Jian

We are trying to calculate the BLEU scores from the best model's outputs. Can you please clarify if you are using "sentence_bleu_score" or "modified_precision" from NLTK? We are getting similar numbers if we use modified_precision. But getting lower numbers if we use "sentence_bleu_score". 

Any help is greatly appreciated.

Thanks
Vinay

**Comment by JianGuanTHU** [comment]
@vinayshekharcmu We implement bleu based on the [open source code]( https://github.com/vikasnar/Bleu). And I am reruning our code and metrices in nltk to make sure we do this right. Thank you!

**Comment by qywu** [comment]
Is there any update on this? A simple generated output should be enough.

**Comment by sairin1202** [comment]
Could you please show the code extracting the relations from the concept Net? Do you use Concept Net5？
